java-compiler-testing/src/main/java/io/github/ascopes/jct/compilers/impl/JctCompilationFactory.java

```diff
@@ -442,14 +383,112 @@ private String determineRelease(JctCompiler<?, ?> compiler) {
+    if (!fileManager.hasLocation(location)) {
+      var dir = workspace.createPath(location);
+      fileManager.addPath(location, dir);
+    }
+  }
+
+  private void configureClassPath(JctFileManagerImpl fileManager) {
+    if (compiler.isInheritClassPath()) {
+      for (var path : jvmClassPath.access()) {
+        var wrapper = wrap(path);
+        fileManager.addPath(StandardLocation.CLASS_PATH, wrapper);
+
+        // IntelliJ appears to place modules on the classpath if we are not building the base
+        // project with JPMS. This is a problem because it means we cannot compile a module
+        // within a test pack not using JPMS, since the modules will be on the classpath rather
+        // than the module path. Fix this by adding classpath components with modules inside into
+        // the module path as well.
+        if (compiler.isFixJvmModulePathMismatch() && containsModules(path)) {
+          fileManager.addPath(StandardLocation.MODULE_PATH, wrapper);
+        }
+      }
+    }
+  }
+
+  private void configureModulePath(JctFileManagerImpl fileManager) {
+    if (compiler.isInheritModulePath()) {
+      for (var path : jvmModulePath.access()) {
+        var wrapper = wrap(path);
+
+        // Since we do not know if the code being compiled will use modules or not just yet,
+        // make sure any modules are on the class path as well so that they remain accessible
+        // in unnamed modules.
+        fileManager.addPath(StandardLocation.CLASS_PATH, wrapper);
+        fileManager.addPath(StandardLocation.MODULE_PATH, wrapper);
+      }
+    }
+  }
+
+  private void configurePlatformClassPath(JctFileManagerImpl fileManager) {
+    if (compiler.isInheritPlatformClassPath()) {
+      for (var path : jvmPlatformPath.access()) {
+        fileManager.addPath(StandardLocation.PLATFORM_CLASS_PATH, wrap(path));
+      }
+    }
+  }
+
+  private void configureJvmSystemModules(JctFileManagerImpl fileManager) {
+    if (compiler.isInheritSystemModulePath()) {
+      for (var path : jvmSystemModules.access()) {
+        fileManager.addPath(StandardLocation.SYSTEM_MODULES, wrap(path));
+      }
+    }
+  }
+
+  private void configureAnnotationProcessorPaths(JctFileManager fileManager) {
+    switch (compiler.getAnnotationProcessorDiscovery()) {
+      case ENABLED:
+        fileManager.ensureEmptyLocationExists(StandardLocation.ANNOTATION_PROCESSOR_PATH);
+        break;
+
+      case INCLUDE_DEPENDENCIES: {
+        // https://stackoverflow.com/q/53084037
+        // Seems that javac will always use the classpath to implement this behaviour, and never
+        // the module path. Let's keep this simple and mimic this behaviour. If someone complains
+        // about it being problematic in the future, then I am open to change how this works to
+        // keep it sensible.
+        fileManager.copyContainers(
+            StandardLocation.CLASS_PATH,
```

Comment:
- [ ] Do I need to do this for the annotation processor module path as well?
- [ ] ...if so, can this be represented as a Map and a loop rather than hardcoding it?